iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/DanglingLineImpl.java

```diff
@@ -122,7 +124,7 @@ public double getTargetQ() {
         @Override
         public GenerationImpl setTargetQ(double targetQ) {
             int variantIndex = danglingLine.network.get().getVariantIndex();
-            ValidationUtil.checkVoltageControl(danglingLine, voltageRegulationOn.get(variantIndex), targetV.get(variantIndex), targetQ);
+            ValidationUtil.checkReactivePowerTarget(this, VALIDABLE_TYPE_DESCRIPTION, targetQ, voltageRegulationOn.get(variantIndex));
```

Comment:
Global remark but I'm still not sure why we want to change the checks in this PR?

Reply:
I reviewed again the validation class trying to improve and simplify it. Additionally to the homogeneity changes, the following changes have been made. These changes are necessary.

Generator controls:
- Do not check `regulatingTerminal` if `useLocalRegulation` is used.
- Fix `checkRegulatingTerminal` as is not considering the enabled attribute.

StaticVarCompensator controls:
- Do not check `regulatingTerminal` if `useLocalRegulation` is used.
- Fix `checkRegulatingTerminal` as is not considering the enabled attribute.

ShuntCompensator controls:
- Do not check `regulatingTerminal` if `useLocalRegulation` is used.
- Fix `checkRegulatingTerminal` as is not considering the enabled attribute.
- Fix `setVoltageRegulatorOn` as the `regulatingTerminal` is not checked.

RatioTapChanger controls:
- Fix `setRegulating` as the `targetDeadband` is not checked.

PhaseTapChanger controls:
- Fix `setRegulating` as the `targetDeadband` is not checked.

Vsc converter controls:
- Do not check `regulatingTerminal` if `useLocalRegulation` is used.

If you want  I can make all the necessary changes trying to discard all the changes associated with homogeneity.